.github/workflows/integration-pacakge-tests.yaml

```diff
@@ -0,0 +1,84 @@
+name: Integrations Packages Tests
+
+on:
+  pull_request:
+    paths:
+      - "src/integrations/*/**"
+  push:
+    branches:
+      - main
+    paths:
+      - "src/integrations/*/**"
+
+jobs:
+  prepare-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.9"
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+          if [[ $GITHUB_EVENT_NAME == 'pull_request' ]]; then
+            COMMIT_RANGE="${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}"
+          else
+            COMMIT_RANGE="${{ github.event.before }}..${{ github.event.after }}"
+          fi
+          python scripts/generate_integration_package_tests_matrix.py "$COMMIT_RANGE" > matrix.json
+          cat matrix.json
+          echo "matrix=$(cat matrix.json)" >> $GITHUB_OUTPUT
```

Comment:
This `prepare-matrix` job ensures that we only run tests for integrations packages with changes.